### ISSUE TITLE
Systemd Notify

### ISSUE THREAD

**Issue by splitice** [gsliepen/tinc#338]
Currently investigating an issue with 1.0 where process is left in "D" (Uninterruptable Sleep) randomly.

Got me thinking that it would be nice if `sd_notify` could be called in the IO loop.

Thoughts?

**Comment by gsliepen** [comment]
Yes, although we don't want to notify systemd every time we call `select()` of course. We can tie it to the periodic checks, like `periodic_handler()` in tinc 1.1 and `check_dead_connections()` in tinc 1.0.

Also, I'd like to avoid needing to link with libsystemd; the protocol is simple enough: check if `$NOTIFY_FD` is set, if so just send `"WATCHDOG=1"` to that fd. We can also use it to notify when tinc has finished initializing and when it's about to quit.

**Comment by splitice** [comment]
Correct, although I'm less keen on re-implementing libsystemd. I'd rather link against the system lib if support is enabled.

Who knows what will change in the future and what hackery people using it might have. Better to use the system supplied library if enabled in my books.

And yes a `timeout_add` for the watchdog makes sense.
